packages/http-signature-utils/src/utils/key.ts

```diff
@@ -1,44 +1,66 @@
 import * as crypto from 'crypto'
 import * as fs from 'fs'
 
-export function parseOrProvisionKey(
-  keyFile: string | undefined
-): crypto.KeyObject {
-  const TMP_DIR = './tmp'
-  if (keyFile) {
-    try {
-      const key = crypto.createPrivateKey(fs.readFileSync(keyFile))
-      if (checkKey(key)) {
-        console.log(`Key ${keyFile} loaded.`)
-        return key
-      } else {
-        console.log('Private key is not EdDSA-Ed25519 key. Generating new key.')
-      }
-    } catch (err) {
-      console.log('Private key could not be loaded.')
-      throw err
-    }
+export function parseKey(keyFilePath: string): crypto.KeyObject {
+  let fileBuffer
+  try {
+    fileBuffer = fs.readFileSync(keyFilePath)
+  } catch (error) {
+    throw new Error(`Could not load file: ${keyFilePath}`)
+  }
+
+  let key
+  try {
+    key = crypto.createPrivateKey(fileBuffer)
+  } catch (error) {
+    throw new Error('File was loaded, but private key was invalid')
+  }
+
+  if (!isKeyEd25519(key)) {
+    throw new Error('Private key did not have Ed25519 curve')
   }
+
+  return key
+}
+
+export function provisionKey(dir: string): crypto.KeyObject {
   const keypair = crypto.generateKeyPairSync('ed25519')
-  if (!fs.existsSync(TMP_DIR)) {
-    fs.mkdirSync(TMP_DIR)
+  if (!fs.existsSync(dir)) {
+    fs.mkdirSync(dir)
   }
   fs.writeFileSync(
-    `${TMP_DIR}/private-key-${new Date().getTime()}.pem`,
+    `${dir}/private-key-${new Date().getTime()}.pem`,
     keypair.privateKey.export({ format: 'pem', type: 'pkcs8' })
   )
   return keypair.privateKey
 }
 
+export function parseOrProvisionKey(keyFile?: string): crypto.KeyObject {
+  if (keyFile) {
+    try {
+      return parseKey(keyFile)
+    } catch (error) {
+      console.warn(
+        `parseOrProvisionKey: could not parse key (${
+          error instanceof Error ? error.message : 'unknown error'
+        }). Generating new key.`
+      )
```

Comment:
This is "basically" the old behaviour, except it will re-generate a key in any error, not just if the key is a different curve.

I'm not sure if I like this behaviour in general, since this fails silently if the key that you expected to load didn't. I'm not sure about the console.logs either in a published package.

Would be great to get some thoughts on this

Reply:
Maybe we should expect the caller to choose between these options:
- parse key (and fail if it cant)
- generate a key (perhaps optional)
- try to parse a key, and regenerate one if it fails

Which we already have as far as I can tell (`parseKey`, `provisionKey`, `parseOrProvisionKey`). I think the function signatures communicate this pretty well already. Maybe we could make the docs for `parseOrProvisionKey` more explicit ("will generate a new key if the parse fails") and include it in a docstring but it's behaving how I would expect personally.

If we really need to communicate parse failures in `parseOrProvisionKey` maybe we return an object like `{key: crypto.KeyObject, parseSucceeded: boolean}`. I guess I'm not really about the console.logs but I feel like we shouldn't be doing it like this. If we wanted to log I imagine we should support passing in popular loggers (winston, bunyon, pino, etc.) and using those (but totally overkill here).

Reply:
Also I dont have the full context on the previous behavior, but distinguishing between a non EdDSA-Ed25519 private key and something unparsable seems like an unimportant distinction to me.

Reply:
Sounds good Blair, agreed. I will add some doc strings as well